README.md

```diff
@@ -1,44 +1,225 @@
-# Open Source Project Template
+# requests_signature_dart
 
-This repository contains a template to seed a repository for an Open Source
-project.
+Signs and validates HTTP requests through Dio interceptors.
 
-## How to use this template
+This projects can help you implements [HMAC](https://en.wikipedia.org/wiki/HMAC) signature to HTTP and HTTPS requests in Dart and/or Flutter.
 
-1. Check out this repository
-2. Delete the `.git` folder
-3. Git init this repository and start working on your project!
-4. Prior to submitting your request for publication, make sure to review the
-   [Open Source guidelines for publications](https://nventive.visualstudio.com/Internal/_wiki/wikis/Internal_wiki?wikiVersion=GBwikiMaster&pagePath=%2FOpen%20Source%2FPublishing&pageId=7120).
+requests_signature_dart depends on 
+- [cryptography](https://pub.dev/packages/cryptography)
+- [dio](https://pub.dev/packages/dio)
+- [uuid](https://pub.dev/packages/uuid)
 
-## Features (to keep as-is, configure or remove)
-- [Mergify](https://mergify.io/) is configured. You can edit or remove [.mergify.yml](/.mergify.yml).
+and is [Dart 3](https://medium.com/dartlang/announcing-dart-3-53f065a10635) compatible.
 
-The following is the template for the final README.md file:
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
 
----
+## Getting Started
 
-# Project Title
+##### Implementing Requests Signature Validation in Dart/Flutter app project:
+
+Install the package via command line:
+
+```
+flutter pub add requests_signature_dart
+```
+
+This will add a line like this to your package's `pubspec.yaml` (and run an implicit `flutter pub get`):
+
+```yaml
+dependencies:
+  request_signature_dart: ^0.0.1
+```
+Alternatively, your editor might support `flutter pub get`. Check the docs for your editor to learn more.
+
+##### Import it 
+
+in your dart code:
+
+```dart
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+```
+
+### Using
+
+#### Implement HMAC signed HTTP requests
+
+```dart
+import 'package:dio/dio.dart'; // For making HTTP requests
+import 'package:requests_signature_dart/requests_signature_dart.dart'; // For request signature functionality
+
+void main() {
+  // Instantiate Dio client
+  final dio = Dio();
+
+  // Define signature options with clockSkew and disableAutoRetryOnClockSkew
+  final signatureOptions = RequestsSignatureOptions(
+    clientId: 'your_client_id', // Your unique client ID
+    clientSecret: 'your_client_secret', // Your client secret key
+    headerName: 'X-Request-Signature', // Name of the custom header for the signature
+    signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}', // Pattern for the signature header value
+    clockSkew: Duration(seconds: 30), // Clock skew duration (30 seconds in this example)
+    disableAutoRetryOnClockSkew: false, // Enable auto retry on clock skew if set to true
```

Comment:
This is confusing because if the value is set to true, then the auto-retry is actually disabled, not enabled. You should update the comment to better align with the parameter name.